lib/xandra/connection/utils.ex

```diff
@@ -48,12 +48,30 @@ defmodule Xandra.Connection.Utils do
     # However, we need to pass the compressor module around when we
     # receive the response to this frame because if we said we want to use
     # compression, this response is already compressed.
-    with :ok <- :gen_tcp.send(socket, payload),
-         {:ok, %Frame{body: <<>>}} <- recv_frame(socket, compressor) do
-       :ok
+    with :ok <- :gen_tcp.send(socket, payload) do
+      case recv_frame(socket, compressor) do
+        {:ok, %Frame{body: <<>>}} -> :ok
```

Comment:
looks like it would be even clearer to write

```
{:ok, %Frame{kind: :ready}} -> :ok
```

but not sure to change the previous behavior